File: Site_Web/covoiturage_la_cite_site_web/core/models/VehicleModel.ts
/**
 * VehicleModel — Modèle unifié pour les véhicules
 * Fusion de : VehiculeModel (domain), VehicleInfo (planner), TripVehicle (trajets)
 */
export interface VehicleModel {
  // ── Identité ──────────────────────────────────────────────────────────────
  id: string;

  // ── Relation ──────────────────────────────────────────────────────────────
  /** ID du propriétaire conducteur (UserModel) */
  driverId: string;

  // ── Informations véhicule ─────────────────────────────────────────────────
  make: string;             // marque, ex: "Honda"
  model: string;            // modèle, ex: "Civic"
  year: number;             // année, ex: 2021
  color: string;            // couleur, ex: "Gris anthracite"
  licensePlate: string;     // immatriculation, ex: "ABC 1234"
  maxSeats: number;         // places max (passagers), ex: 4

  // ── Ressources visuelles ──────────────────────────────────────────────────
  photoUrl?: string;

  // ── Statut ────────────────────────────────────────────────────────────────
  /** Véhicule actif (utilisé pour les trajets) */
  isActive: boolean;
  /** Véhicule validé par l'administration */
  isValidated: boolean;

  // ── Metadata ──────────────────────────────────────────────────────────────
  createdAt: string;
  updatedAt: string;
}

File: App_Mobile/Covoiturage_la_cite_(App_Mobile)/Covoiturage_la_cite_(App_Mobile)/Core/Models/VehicleModel.cs
// ============================================================
//  Core/Models/VehicleModel.cs
//  Véhicule du conducteur (formulaire création trajet)
// ============================================================

namespace Covoiturage_la_cite__App_Mobile_.Core.Models
{
    public class VehicleModel
    {
        public string Id { get; init; } = "";
        public string DriverId { get; init; } = "";
        public string Make { get; init; } = "";
        public string Model { get; init; } = "";
        public int? Year { get; init; }
        public string? Color { get; init; }
        public string? LicensePlate { get; init; }
        public int MaxSeats { get; init; }
        public bool IsActive { get; init; }
        public bool IsValidated { get; init; }

        /// <summary>Label affiché dans le sélecteur (ex : "Honda Civic 2020")</summary>
        public string Label => $"{Make} {Model}{(Year.HasValue ? $" {Year}" : "")}";

        /// <summary>Nombre max de passagers = MaxSeats - 1 (siège conducteur)</summary>
        public int MaxPassengers => Math.Max(0, MaxSeats - 1);
    }
}
